docs/docs/08.1-more-about-refs.md

```diff
@@ -86,75 +87,40 @@ It's as simple as:
+```javascript
+var MyComponent = React.createClass({
+  handleClick: function() {
+    // Explicitly focus the text input using the raw DOM API.
+    this.myTextInput.focus();
   },
-```
-
-or
-
-```html
   render: function() {
+    // The ref attribute adds a reference to the component to
+    // this.refs when the component is mounted.
     return (
-      <TextInput
-        ref={function(input) {
-          if (input != null) {
-            input.focus();
-          }
-        }} />
+      <div>
+        <input type="text" ref="(ref) => this.myTextInput = ref" />
+        <input
+          type="button"
+          value="Focus the text input"
+          onClick={this.handleClick}
+        />
+      </div>
     );
-  },
-```
-
-Note that when the referenced component is unmounted and whenever the ref changes, the old ref will be called with `null` as an argument. This prevents memory leaks in the case that the instance is stored, as in the first example. Note that when writing refs with inline function expressions as in the examples here, React sees a different function object each time so on every update, ref will be called with `null` immediately before it's called with the component instance.
-
-
-## Completing the Example
+  }
+});
 
-```javascript
-  var App = React.createClass({
-    getInitialState: function() {
-      return {userInput: ''};
-    },
-    handleChange: function(e) {
-      this.setState({userInput: e.target.value});
-    },
-    clearAndFocusInput: function() {
-      // Clear the input
-      this.setState({userInput: ''}, function() {
-        // This code executes after the component is re-rendered
-        React.findDOMNode(this.refs.theInput).focus();   // Boom! Focused!
-      });
-    },
-    render: function() {
-      return (
-        <div>
-          <div onClick={this.clearAndFocusInput}>
-            Click to Focus and Reset
-          </div>
-          <input
-            ref="theInput"
-            value={this.state.userInput}
-            onChange={this.handleChange}
-          />
-        </div>
-      );
-    }
-  });
+React.render(
+  <MyComponent />,
+  document.getElementById('example')
+);
 ```
 
-In this example, our render function returns a description of an `<input />` instance. But the true instance is accessed via `this.refs.theInput`. As long as a child component with `ref="theInput"` is returned from `render`, `this.refs.theInput` will access the proper instance. This even works on higher level (non-DOM) components such as `<Typeahead ref="myTypeahead" />`.
+In this example, we get a reference to the text input component and we call `focus()` on the component when the button is clicked.
 
+For composite components, the reference will refer to an instance of the component class so you can invoke any methods that are defined on that class.  If you need access to the underlying DOM node for that component, you can use [findDOMNode](/react/docs/top-level-api.html#react.finddomnode).
```

Comment:
May as well be consistent with above where you use the full API: `React.findDOMNode()`